File: corutine/test-1.py
# test-1.py
# 定义生成器
def consume():
    print("enter consume")
    while True:
        # consumer 协程等待接收数据
        number = yield
        print("consumer: Get", number)
# 创建协程对象
consumer = consume()
print("created consumer")
# 启动协程
next(consumer)
print("after next")
for num in range(3):
    print("Create", num)
    consumer.send(num)

File: corutine/test-1.c
// test-1.c
#include <stdio.h>

void consume(int number){
    static int state = 0;
    switch(state){
        case 0:
            printf("enter consume\n");
            state++;
            while(1){
                printf("consumer: Get %d\n", number);
                return;
        case 1: continue;
            }
    }
}
int main(){
    for(int i=0; i<3; ++i) {
        printf("Create %d\n", i);
        consume(i);
    }
}

